core/capabilities/ccip/ccipsolana/helpers.go

```diff
@@ -0,0 +1,32 @@
+package ccipsolana
+
+import (
+	"fmt"
+	"reflect"
+
+	agbinary "github.com/gagliardetto/binary"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_router"
+)
+
+// DecodeExtraArgsToMap is a helper function for converting Borsh encoded extra args bytes into map[string]any, which will be saved in ocr report.message.ExtraArgsDecoded
+func DecodeExtraArgsToMap(extraArgs []byte) (map[string]any, error) {
```

Comment:
Previously, the EVM version had this comment and a special offset. What happened to it?
```
	case string(svmExtraArgsV1Tag):
		// for SVMExtraArgs there's the four bytes plus another 32 bytes padding for the dynamic array
		method = svmV1DecodeName
		extraByteOffset = 36
```